src/main/java/net/snowflake/ingest/streaming/internal/ParquetChunkData.java

```diff
@@ -35,16 +34,6 @@ public ParquetChunkData(
     this.rows = rows;
     this.parquetWriter = parquetWriter;
     this.output = output;
-    // create a defensive copy of the parameter map because the argument map passed here
-    // may currently be shared across multiple threads.
-    this.metadata = createDefensiveCopy(metadata);
```

Comment:
is this fix not needed any more, under high concurrent load?

Reply:
technically no, as all the other fields in this map will stay the same across all files (for now). But I reverted this back to keep the logic less error-prone in the future.